Vault default parallelism with Consul backend exceeds new Consul 1.6.3 default limits

#### Overview of the Issue

Since #7159, there's now a default limit of 100 simultaneous HTTP connections coming from a single IP.  When Vault is using Consul as a backend, by default its [max_parallel](https://www.vaultproject.io/docs/configuration/storage/consul/#inlinecode-max_parallel-13) setting allows it to open up to 128 connections to Consul.  When Vault exceeds the connection limit, it sees it as a backend failure, leading it to sealing itself.

#### Reproduction Steps

Steps to reproduce this issue:

I haven't tried to reproduce it directly, though it's my analysis of logs after a failure that led me to conclude this is the problem.  I expect it could be reproduced by inducing a high number of parallel requests to Vault with the default Consul/Vault settings.  When I observed it was during the lease restoration phase of Vault startup, which runs 64 parallel requests.  I imagine there must have been additional traffic to account for the remaining 37 requests that would be needed to exceed the limit of 100, but I don't know yet where that's coming from, since Vault isn't unsealed while the restoration is ongoing.  Either way I'm confident that this new 1.6.3 behaviour will be problematic for some existing Vault+Consul users.

### Consul info for both Client and Server

<details>
  <summary>Client info</summary>

Vault 1.2.3+ent.

</details>

<details>
  <summary>Server info</summary>

I don't have direct access to the system in question, but this was observed multiple times with Consul 1.6.3, and does not manifest with Consul 1.6.1.

</details>

### Operating system and Environment details

Best I can do right now is "linux, amd64".

### Log Fragments

```
vault: 2020-02-11T12:46:53.984-0600 [ERROR] expiration: error restoring leases: error="failed to read lease entry XXXX: Get http://127.0.0.1:8500/v1/kv/vault/sys/expire/id/XXXX: read tcp 127.0.0.1:50460->127.0.0.1:8500: read: connection reset by peer"
```

```
vault: 2020-02-11T12:38:36.634-0600 [ERROR] expiration: error restoring leases: error="failed to read lease entry YYYY: Get http://127.0.0.1:8500/v1/kv/vault/sys/expire/id/YYYY: EOF"
```

--- a/agent/config/default.go
+++ b/agent/config/default.go
@@ -103,7 +103,7 @@ func DefaultSource() Source {
 			recursor_timeout = "2s"
 		}
 		limits = {
-			http_max_conns_per_client = 100
+			http_max_conns_per_client = 200
 			https_handshake_timeout = "5s"
 			rpc_handshake_timeout = "5s"
 			rpc_rate = -1

--- a/agent/config/runtime.go
+++ b/agent/config/runtime.go
@@ -806,7 +806,7 @@ type RuntimeConfig struct {
 	// HTTPMaxConnsPerClient limits the number of concurrent TCP connections the
 	// HTTP(S) server will accept from any single source IP address.
 	//
-	// hcl: limits{ http_max_conns_per_client = 100 }
+	// hcl: limits{ http_max_conns_per_client = 200 }
 	HTTPMaxConnsPerClient int
 
 	// HTTPSHandshakeTimeout is the time allowed for HTTPS client to complete the

--- a/agent/config/runtime_test.go
+++ b/agent/config/runtime_test.go
@@ -3466,7 +3466,7 @@ func TestConfigFlagsAndEdgecases(t *testing.T) {
 				// intentional.
 				rt.RPCHandshakeTimeout = 5 * time.Second
 				rt.HTTPSHandshakeTimeout = 5 * time.Second
-				rt.HTTPMaxConnsPerClient = 100
+				rt.HTTPMaxConnsPerClient = 200
 				rt.RPCMaxConnsPerClient = 100
 			},
 		},

--- a/website/source/docs/agent/options.html.md
+++ b/website/source/docs/agent/options.html.md
@@ -1397,7 +1397,7 @@ Valid time units are 'ns', 'us' (or 'µs'), 'ms', 's', 'm', 'h'."
         Configures a limit of how many concurrent TCP connections a single
         client IP address is allowed to open to the agent's HTTP(S) server. This
         affects the HTTP(S) servers in both client and server agents. Default
-        value is `100`.
+        value is `200`.
     *   <a name="https_handshake_timeout"></a><a
         href="#https_handshake_timeout">`https_handshake_timeout`</a> -
         Configures the limit for how long the HTTPS server in both client and

